xarray/core/dataarray.py

```diff
@@ -2663,8 +2663,15 @@ def func(self, other):
             # don't support automatic alignment with in-place arithmetic.
             other_coords = getattr(other, "coords", None)
             other_variable = getattr(other, "variable", other)
-            with self.coords._merge_inplace(other_coords):
-                f(self.variable, other_variable)
+            try:
+                with self.coords._merge_inplace(other_coords):
+                    f(self.variable, other_variable)
+            except MergeError:
+                raise ValueError(
```

Comment:
Can we still raise `MergeError`? It is possible some users are relying on the exception type.

Note that MergeError is a subclass of ValueError